govcd/api_vcd_versions.go

```diff
@@ -178,3 +178,12 @@ func (cli *Client) validateAPIVersion() error {
 
 	return nil
 }
+
+// GetDefaultAPIVersionOrLaterThan32 returns default version or 32 if it is connected to version vCD 9.7 or later
```

Comment:
I don't understand the reason for this function, with an API version hard coded in the name.
* What's the problem we need to solve?
* Can this function become a more general one, so that it doesn't become obsolete when 32 is not so important?

Reply:
Reason:
If vCD 9.7 or later we need to use 32.0 version in some requests to work with internal disk in VM.
I could copy paste these 4 lines, but created function to be nice. My aim to create specific function for dedicated version override. Not generic one.

Reply:
I guess @dataclouder  meant something like `func (cli *Client) GetDefaultAPIVersionOrLaterThan(32) string`

Reply:
Well, if you both want that I could change function. My intention was to have explicit traceability by version in which function we override. Also hides need to know details how to define exact API version. As for example it's not `32` it's `32.0`

Reply:
We really need a generalized function with the purpose of having a **single code path** for dealing with version manipulations. Possible name suggestion:
```
func (cli *Client) GetSpecificApiVersionOnCondition(vcdApiVersionCondition, wantedApiVersion)
```
Call example:
```
client.GetSpecificApiVersionOnCondition(">= 32.0", "32.0")
```
The reason I propose to expose the ">=" in addition to the number, because it's actually clearer of what the function does with this approach than to interpret a very long function name.

For example, current name "GetDefaultAPIVersionOrLaterThan32" is very hard to grasp and is also not precise, because in reality it is "GetDefaultApiVersionOr32IfVersionIsLaterThan32". And at this point this conundrum reminds me of "42" and the true meaning of life :)

BTW, "API" should be "Api" ;)

Of course, in this case, the comment needs to state that the default version is returned if condition is not met.

Reply:
done